static/src/javascripts/projects/commercial/modules/sticky-mpu.js

```diff
@@ -35,19 +37,15 @@ define([
             var options = config.page.isAdvertisementFeature ? {top: 43} : {};
             stickyElement = new Sticky($adSlot[0], options);
             stickyElement.init();
-            messenger.register('set-ad-height', onAppNexusResize);
+            messenger.register('resize', onResize);
             return stickyElement;
         });
     }
 
-    function onAppNexusResize(specs, _, iframe) {
-        var adSlot = closest(iframe, '.js-ad-slot');
-        if (adSlot.id === rightSlotId) {
-            messenger.unregister('set-ad-height', onAppNexusResize);
-            fastdom.write(function () {
-                iframe.style.height = specs.height + 'px';
-                stickyElement.updatePosition();
-            });
+    function onResize(specs, _, iframe) {
+        if (rightSlot.contains(iframe)) {
```

Comment:
In the [resize](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/messenger/resize.js) handler 😄 
Now both sticky-top and sticky-mpu can focus on making sure the stickiness works correctly. There is a bit of duplication now because Rubicon ads resize the iframe, but eventually AdOps can update the code in DFP.